src/main/java/com/yelp/nrtsearch/server/luceneserver/StatsRequestHandler.java

```diff
@@ -60,14 +60,16 @@ private StatsResponse process(IndexState indexState) throws IOException {
         for (Map.Entry<Integer, ShardState> entry : indexState.shards.entrySet()) {
             ShardState shardState = entry.getValue();
             statsResponseBuilder.setOrd(entry.getKey());
-            IndexWriter.DocStats docStats = shardState.writer.getDocStats();
+            if (shardState.writer != null) { //replica
```

Comment:
does this mean that maxDoc and numDocs would be available only for replicas ?

Reply:
correct

Reply:
can we have an `else` or `else if` that follows the previous behavior so that we get maxDoc and numDocs with primary and standalone nodes as well ?

Reply:
sorry the comment is misleading, the code should work already for both replicas and standalone. `writer` is null only for replicas and hence we cannot access the DocStats

Reply:
fixed the comment